### ISSUE TITLE
Heterozygous peak identified as errors?

### ISSUE THREAD

**Issue by Haoran-Xue** [schatzlab/genomescope#131]
Hello,

I ran kmc and kmc_tools with PacBio HiFi sequences of a diploid plant species:
kmc -m128 -k21 -t40 -ci1 -cs10000 xxx.hifi.fastq.gz xxx xxx_tmp
kmc_tools transform xxx histogram xxx.histo

Then I submit the histo file to GenomeScope2.0 (http://genomescope.org/genomescope2.0/), with "K-mer length: 21, Ploidy: 2, Max k-mer coverage: -1, Average k-mer coverage for polyploid genome: -1".

This is the linear plot I got:

![linear_plot](https://github.com/schatzlab/genomescope/assets/39385521/b35df2a0-e635-4a67-99ff-99c773a70186)

It seems that the fist peak (heterozygous peak) was identified as errors. Is there any way to avoid this?

Thank you!

**Comment by fperezcobos** [comment]
Hi,

I had the same problem, PacBio HiFi sequences of a diploid plant species and the plot looks like this:

![Brassica_linear_plot](https://github.com/schatzlab/genomescope/assets/154519851/71da8eec-52cb-4d2e-b531-3c7c157a48da)

Any help?

**Comment by SamCT** [comment]
Also seeing this with one of our genomes. Of a lot of four Revio SMRT cells (all the same species) one plot looks like the above. This particular SMRT cell that has this plot has a much higher number of reads than others, but besides that nothing stands out. The other three plots looked reasonable. I'm curious to know what is causing this

**Comment by mschatz** [comment]
The automatic model fitting algorithm can get confused if you have too high
of coverage or if there is ambiguity in the relationships between the
homozygous and heterozygous peaks. The easiest way to address is to use the
"Average k-mer coverage for polyploid genome" parameter which gives a hint
as to where the first peak (heterozygous peak) is located. For these
datasets I would try with a value of about 100. If that doesnt work, the
next easiest thing to do is downsample the read dataset to reduce the
coverage. From a raw read file, you can just use 'head' to select the first
N lines in the file to reduce the number of reads, which serves as a random
downsample (assuming the reads have not been aligned or other processing
has happened)

Good luck!

Mike

**Comment by xiaoyu-stars** [comment]
I also encountered this problem, and my solution was to use the “-l” parameter to specify the position of my heterozygous peak such as -l 29(this is my setting number). Actually, I’m not sure whether this counts as solving the problem, because I don’t fully understand the underlying principle, but doing this made my results better (the genome size became more accurate, and the heterozygosity rate returned to normal).

**Comment by mschatz** [comment]
Yes, the "-l" parameter for the command line version is the same as the
"Average k-mer coverage for polyploid genome" for the web version. Either
way it gives the algorithm a hint as to which peak should be used as the
baseline for estimating the genome size and heterozygosity rate

Good luck!

Mike

**Comment by Haoran-Xue** [comment]
Hi Mike,

Setting the "Average k-mer coverage for polyploid genome" parameter worked. Thank you so much!

<img width="2000" height="2000" alt="Image" src="https://github.com/user-attachments/assets/77bf70b4-94c6-404f-b55f-ea61b705f5aa" />

Best regards,
Haoran
